### ISSUE TITLE
Fallback policy not triggering for retrival intents or sub intents

### ISSUE THREAD

**Issue by Snorlaxcode** [RasaHQ/rasa#9723]
### What problem are you trying to solve?

Hi

I am unable to trigger the fallback policy for rasa retrieval intents or sub-intents of the main faq class.

I’ve 220 faq questions defined as sub intents of the main intent faq. When some random questions such as ‘How is the weather or how is life going or shall I invest in cryptos?’ are asked, the system defaults to faq intent with 0.99 probability.

Afterward, it picks a random faq answer (sub intent/retrieval intent) (with the highest confidence of ~0.1) and throws back its response. I was able to ascertain this by running rasa in debug and nlu mode.

nlu and domain example:

intent: faq/FAQ_ID_08 examples: |

Asset status when workorder is raised for lost tool utter_faq/FAQ_ID_08:

text: 'The workorder status will be UNDER TOOLS SEARCH until the tool is found. ’ Domain:

utter_please_rephrase:

text: I’m sorry, I didn’t quite understand that. Could you rephrase? text: I didn’t quite get it. Can you put it in different words? text: Apologies, I did not understand. Can you ask in again in different words? Rules:

rule: Ask the user to rephrase whenever they send a message with low NLU confidence steps: intent: nlu_fallback action: utter_please_rephrase Config:

name: FallbackClassifier threshold: 0.3 ambiguity_threshold: 0.1 Rasa version: 2.4.3

### What's your suggested solution?

The system needs to treat sub-intents or retrieval the same as regular intents while checking intent confidence scores (matching from the existing list of intents) instead of grouping in FAQ singular intent.

The fallback classifier needs to treat each sub-intent as a separate intent while triggering a fallback policy instead of treating faq as a singular intent encompassing all 220 (or more) sub-intents. In an faq heavy chatbot with way more faqs than regular intents, it leads to faq having a confidence score of 0.99 while others including fallback have negligible confidence.

### Examples (if relevant)

[https://forum.rasa.com/t/fallback-policy-not-triggering-for-retrival-intents-or-sub-intents/45504/2](https://forum.rasa.com/t/fallback-policy-not-triggering-for-retrival-intents-or-sub-intents/45504/2)

### Is anything blocking this from being implemented? (if relevant)

 _No response_ 

### Definition of Done

If in a >200 sub intent containing system with <10 other intents, the bot is able to trigger the fallback classifier when asked out of context questions.

**Comment by PhilCorcoran** [comment]
Have the same problem with a FAQ and would like to see this implemented

**Comment by sara-tagger** [comment]
Thanks for submitting this feature request 🚀 @amn41 will get back to you about it soon! ✨

**Comment by nbeuchat** [comment]
That's a great point, I often encounter the same problem where many sub-intents have about the same probability. I don't think the number of sub-intent should matter though.

One workaround you can do (but it's a bit painful and manual) is to create a custom action to answer to this faq and you manually check the confidence of the top N sub-intents that were picked. If you're confident that it's the right sub-intent, you can utter the response and if not, trigger the fallback. It's obviously not ideal and I'd much better have a configurable solution as suggested here!

**Comment by Snorlaxcode** [comment]
@amn41 is there any updates on this?

**Comment by jsalbr** [comment]
I have the same issue and would like to see a solution implemented.

**Comment by wenkang-du** [comment]
> I have the same issue and would like to see a solution implemented.

+1
